src/lsm/table.zig

```diff
@@ -21,8 +21,13 @@ pub const TableUsage = enum {
     /// If your usage fits this pattern:
     /// * Only put keys which are not present.
     /// * Only remove keys which are present.
-    /// * TableKey == TableValue (modulo padding, eg CompositeKey)
-    /// Then we can unlock additional optimizations.
+    /// * TableKey == TableValue (modulo padding, eg CompositeKey).
+    /// Then we can unlock additional optimizations:
+    /// * Immediately cancel out tombstone  and the corresponding insert, without waiting for the
+    ///   tombstone to sink to the bottom of the LSM true: absence of upserts guarantees that
```

Comment:
I don't quite follow the "absence of upserts guarantees" bit? Do you mean because we explicitly know if we're inserting vs updating we can guarantee that if we have a pair that cancels, it couldn't possibly have existed on a lower level?

Reply:
Good point, I think this should say just "updates" --- what we don't have here is to consequitive inserts, two inserts are always separated by a tombstone.